src/map/utils/charutils.cpp

```diff
@@ -3325,10 +3325,14 @@ namespace charutils
 
     void CheckWeaponSkill(CCharEntity* PChar, uint8 skill)
     {
-        auto* weapon = dynamic_cast<CItemWeapon*>(PChar->m_Weapons[SLOT_MAIN]);
+        auto* weapon       = dynamic_cast<CItemWeapon*>(PChar->m_Weapons[SLOT_MAIN]);
+        auto* rangedWeapon = dynamic_cast<CItemWeapon*>(PChar->m_Weapons[SLOT_RANGED]);
         if (!weapon || weapon->getSkillType() != skill)
         {
-            return;
+            if (!rangedWeapon || rangedWeapon->getSkillType() != skill)
```

Comment:
Should/could these two sets of conditions be condensed instead of nested?
```cpp
    auto* weapon       = dynamic_cast<CItemWeapon*>(PChar->m_Weapons[SLOT_MAIN]);
    auto* rangedWeapon = dynamic_cast<CItemWeapon*>(PChar->m_Weapons[SLOT_RANGED]);

    bool noOrInvalidMainWeapon = !weapon || weapon->getSkillType() != skill;
    bool noOrInvalidRangedWeapon = !rangedWeapon || rangedWeapon->getSkillType() != skill;

    if (noOrInvalidMainWeapon && noOrInvalidRangedWeapon)
    {
        return;
    }
```

Reply:
Makes sense, done.